lib/core/src/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobin.hs

```diff
@@ -114,52 +114,66 @@ import qualified Data.Set as Set
 --
 data SelectionCriteria = SelectionCriteria
     { outputsToCover
-        :: NonEmpty TxOut
+        :: !(NonEmpty TxOut)
     , utxoAvailable
         :: !UTxOIndex
+    , extraCoinSource
+        :: !(Maybe Coin)
     }
     deriving (Eq, Show)
 
 -- | The result of performing a successful selection.
 --
-data SelectionResult = SelectionResult
+data SelectionResult (change :: *) = SelectionResult
     { inputsSelected
         :: !(NonEmpty (TxIn, TxOut))
+        -- ^ A (non-empty) list of selected inputs from the wallet's UTxO.
+    , outputsConsidered
+        :: !(NonEmpty TxOut)
+        -- ^ Users aren't typically aware of the minimum coin value needed for
+        -- each output. So, we do alter any requested non-ada output to include
+        -- a minimum ada value.
     , changeGenerated
-        :: !(NonEmpty TokenBundle)
-    , utxoRemaining
-        :: !UTxOIndex
```

Comment:
I did move it aside so that we could re-use the `SelectionResult` type for the fee calculation function. The fee calculation would do nothing with the leftover UTxO so it'd be misleading to pass it around. 

Though, I ended up having to tweak the `SelectionResult` to also introduce a type-parameter which maybe is a sign that in the end, we simply need a different type altogether.

Reply:
> I did move it aside so that we could re-use the SelectionResult type for the fee calculation function. The fee calculation would do nothing with the leftover UTxO so it'd be misleading to pass it around.
> Though, I ended up having to tweak the SelectionResult to also introduce a type-parameter which maybe is a sign that in the end, we simply need a different type altogether.

I agree it's probably a good idea to use a separate type internally for the fee calculation, and then keep the `SelectionResult` as a simple record type without a type parameter. 👍🏻 

It certainly makes it easier for callers of `performSelection` to understand, I think!